src/xenia/app/emulator_window.cc

```diff
@@ -928,6 +936,41 @@ void EmulatorWindow::ToggleFullscreen() {
   SetFullscreen(!window_->IsFullscreen());
 }
 
+void EmulatorWindow::TakeScreenshot() {
+  xe::ui::RawImage myimage;
+  if (emulator()->is_title_open()) {  // Make sure we have a game open first
+    GetGraphicsSystemPresenter()->CaptureGuestOutput(myimage);
+    ExportImage(myimage);
+  }
+}
+
+void EmulatorWindow::ExportImage(const xe::ui::RawImage& image) {
+  auto path = std::filesystem::path("screenshot.png");
```

Comment:
It may be desirable to save the screenshot somewhere in a more user-faced directory. The best default location would likely be a subdirectory (preferably something like `Xenia/game name` with characters not supported in paths on the host file system filtered out, or `Xenia/title ID` if there's no game name provided in the XEX) in the user images directory in the OS. Other non-default options may be a directory specified in the config, and the executable directory in case the emulator was launched in the portable mode.